plugins/home/src/homePageComponents/WelcomeTitle/WelcomeTitle.tsx

```diff
@@ -13,20 +13,40 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import { identityApiRef, useApi } from '@backstage/core-plugin-api';
+import {
+  alertApiRef,
+  identityApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
 import { Tooltip } from '@material-ui/core';
-import React, { useMemo } from 'react';
+import React, { useEffect, useMemo } from 'react';
+import { useAsync } from 'react-use';
 import { getTimeBasedGreeting } from './timeUtil';
 
 export const WelcomeTitle = () => {
   const identityApi = useApi(identityApiRef);
-  const profile = identityApi.getProfile();
-  const userId = identityApi.getUserId();
+  const alertApi = useApi(alertApiRef);
   const greeting = useMemo(() => getTimeBasedGreeting(), []);
 
+  const { value: user, error } = useAsync(async () => ({
+    profile: await identityApi.getProfileInfo(),
+    identity: await identityApi.getBackstageIdentity(),
+  }));
+
+  useEffect(() => {
+    if (error) {
+      alertApi.post({
+        message: `Failed to load user identity: ${error}`,
+        severity: 'error',
+      });
+    }
+  }, [error, alertApi]);
+
   return (
     <Tooltip title={greeting.language}>
-      <span>{`${greeting.greeting}, ${profile.displayName || userId}!`}</span>
+      <span>{`${greeting.greeting}, ${
+        user?.profile.displayName || user?.identity.userEntityRef
```

Comment:
I dunno, this is such a personal greeting - in this component, maybe we should give up entirely if there is no nice display name, and just end with an exclamation mark instead :) "Good Evening!"

Reply:
Otherwise I think it might end up sounding a bit like something out of Futurama - "Good Evening, *robotic voice* DEFAULT USER UUID 153235424!" 😆

Reply:
makes sense, I'll update it